### PR TITLE
add rxcpp

### PR DESCRIPTION
**Pull request by paul-reilly**
add rxcpp header only lib.

### CODE REVIEW

**Comment by paul-reilly** [comment]
I don't know how to fix apple clang erroring on the trailing return type or the braced initializer in the CI reports.

File: packages/r/rxcpp/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    add_versions("v4.1.1", "90758767f7299ff94ba7f359944fde479a7aeb92")
+    add_versions("v3.0.0", "8290f92f744f807e83b1bfe9e8c0ffd162140ec8")
+
+    on_install(function(package)
+        os.cp("Rx/v2/src/rxcpp", package:installdir("include"))
+    end)
+
+    on_test(function(package)
+        assert(package:check_cxxsnippets({test = [[
+            #include <rxcpp/rx.hpp>
```
Comment (waruqi) [inline packages/r/rxcpp/xmake.lua:18]: you need add some class/api call code. example code.

File: packages/r/rxcpp/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        ]]}, { configs = {languages="c++11"}}))
```
Comment (waruqi) [inline packages/r/rxcpp/xmake.lua:19]: add spaces ` = `

File: packages/r/rxcpp/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package("rxcpp")
+
+    set_kind("library", { headeronly = true })
+    set_homepage("http://reactivex.io/RxCpp/")
+    set_description("RxCpp is an implementation of the Observable/Observer pattern and a set of operations. It is an async equivalent of the container/iterator pattern and the set of algorithms found in the STL.")
```
Comment (waruqi) [inline packages/r/rxcpp/xmake.lua:5]: too long description

File: packages/r/rxcpp/xmake.lua (hunk 2 above)
Comment (paul-reilly) [reply packages/r/rxcpp/xmake.lua:19]: I've run the package with the new test through stylua to format it.

**Pull request merged**